code/modules/projectiles/guns/energy.dm

```diff
@@ -20,6 +20,18 @@
 	var/use_cyborg_cell = FALSE //whether the gun's cell drains the cyborg user's cell to recharge
 	var/dead_cell = FALSE //set to true so the gun is given an empty cell
 
+/// Creates an energy ammo case and various projectiles in order to retrieve combat information
+/obj/item/gun/energy/proc/get_notes()
+	if(!ammo_type)
+		return
+	/// Output added to the offensive notes variable
+	var/notebuilder = ""
+	for(var/ammo_obj in ammo_type)
+		var/obj/item/ammo_casing/energy/pew = new ammo_obj(src) /// Need to make a new case...
+		var/obj/projectile/pew_beam = new pew.projectile_type /// ...And laser, for every type
+		notebuilder += uppertext("[pew.select_name]: [pew_beam.damage] [pew_beam.damage_type]\n")
+	return notebuilder
```

Comment:
Oh also, it's better to use lists and Join() <https://secure.byond.com/docs/ref/#/list/proc/Join> for this sort of thing, because of the inefficiencies of adding to strings. it's not a big thing, but it's consistent with how other procs of the same type work, so you should do the same.